uefi/src/proto/misc.rs

```diff
@@ -23,3 +26,56 @@ impl Timestamp {
         unsafe { (self.0.get_properties)(&mut properties) }.to_result_with_val(|| properties)
     }
 }
+
+/// Protocol to register for a notification when ResetSystem is called.
+#[derive(Debug)]
+#[repr(transparent)]
+#[unsafe_protocol(ResetNotificationProtocol::GUID)]
+pub struct ResetNotification(ResetNotificationProtocol);
+
+impl ResetNotification {
+    /// Register a notification function to be called when ResetSystem() is called.
+    ///
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use log::info;
+    /// use uefi::Handle;
+    /// use uefi::prelude::BootServices;
+    /// use uefi::proto::misc::{ResetNotification};
+    /// use uefi_raw::Status;
+    /// use uefi_raw::table::runtime;
+    ///
+    ///
+    /// /* value efi_reset_fn is the type of ResetSystemFn, a function pointer*/
+    ///     unsafe extern "efiapi" fn efi_reset_fn(
```

Comment:
Remove four spaces from the beginning of each of these lines so that the code isn't extra indented.